screens/teachers/TeacherProfileScreen.js

```diff
@@ -4,105 +4,114 @@ import { styles } from '../../styles/styles';
-      return(
-        <View key={i} style={teacherStyles.base}>
-          <View style={teacherStyles.div_1}>
-            <View style={teacherStyles.div_2}>
-            <Text style={textStyles.titleLarge}>
-            {teacher.first_name} {teacher.last_name}
-            </Text>
-            </View>
-
-            <View style={teacherStyles.div_2}>
-            <Text style={textStyles.titleSmall}>
-            Dream ID
-            </Text>
-            <Text style={textStyles.body}>
-            {teacher.dream_id}
-            </Text>
-            </View>
-
-            <View style={teacherStyles.div_2}>
-            <Text style={textStyles.titleSmall}>
-            Email
-            </Text>
-            <Text style={textStyles.body}>
-            {teacher.email}
-            </Text>
-            </View>
-
-            <View style={teacherStyles.div_2}>
-            <Text style={textStyles.titleSmall}>
-            Phone Number
-            </Text>
-            <Text style={textStyles.body}>
-            {teacher.phone}
-            </Text>
-            </View>
+  _renderTeacher() {
+    return(
+      <View style={teacherStyles.base}>
+        <View style={teacherStyles.div_1}>
+          <View style={teacherStyles.div_2}>
+          <Text style={textStyles.titleLarge}>
+          {this.state.teacher.first_name} {this.state.teacher.last_name}
+          </Text>
+          </View>
+
+          <View style={teacherStyles.div_2}>
+          <Text style={textStyles.titleSmall}>
+          Dream ID
+          </Text>
+          <Text style={textStyles.body}>
+          {this.state.teacher.dream_id}
+          </Text>
+          </View>
+
+          <View style={teacherStyles.div_2}>
+          <Text style={textStyles.titleSmall}>
+          Email
+          </Text>
+          <Text style={textStyles.body}>
+          {this.state.teacher.email}
+          </Text>
+          </View>
+
+          <View style={teacherStyles.div_2}>
+          <Text style={textStyles.titleSmall}>
+          Phone Number
+          </Text>
+          <Text style={textStyles.body}>
+          {this.state.teacher.phone}
+          </Text>
           </View>
         </View>
-      );
-    });
+      </View>
+    );
   }
 
   render() {
     const { navigate } = this.props.navigation;
     let teachers;
     if (this.state.isLoading) {
-      teachers = (
+      teacher = (
         <Text>Loading...</Text>
       )
     } else {
-      teachers = this._renderTeachers()
+      teacher = this._renderTeacher()
     }
     return (
       <ScrollView>
         <View>
-          { teachers }
+          { teacher }
+
+          <Button
+             title="Edit Profile" onPress={()=>{ navigate('EditTeacherProfile', 
+              { refreshTeacher: this._fetchTeacher, teacher: this.state.teacher }); }} 
+          />
         </View>
       </ScrollView>
     );
 
   }
+
+  static navigationOptions = ({navigation}) => ({
```

Comment:
In the future, we might want to figure out how to keep navigations options in routes but still give it state from the current screen